Principal page of thenextpage project

sipe-daniel opened the pull request:
This project consist to build a page which matches with the main page of thenextprage website

**_I received permission to work alone on this project_**

sipe-daniel committed:
feat(principal-page): add some html codes

sipe-daniel committed:
refactor(readme-file): add some codes in order to improve the description of this project

sumancrest0001 reviewed (commented):
**Status: changes required
Hello @sipe-daniel,**
You have done a good job but we have to give a little more effort and make some improvements to get approval for this project. Let's fix the following issues.
**[NOTE] All the requested changes are mandatory.**

 Please add a title and a short description of the features added.

It seems like you forgot to enable the stickler for this project. Can you please enable it and fix error if any.

sumancrest0001 commented on index.html:
```diff
@@ -1,9 +1,543 @@
+            
+        </header>
+        <main class="wrapper">
+            <!-- <div class="presentation">
+                <article class="cover-image firstpresetation-bloc ">
+                    <h1>This hybrid AI system can <br/> understand causality in controlled<br/> environments</h1>
+                    
+                    <p>by <a href="#" class="link-white">Ben Dickson</a><i class="far fa-clock"></i> 6 hours ago <i class="fas fa-share-alt"></i> 0
+                    </p>
+                </article>
+                <div class="secondpresetation-bloc cover-trigger">
+                    <article>
+                        <h1>Shopify, Basecamp, and more will<br/>discuss the future of ecommerce<br/>at Sprint on June 11</h1>
+                    </article>
+                    <article>
+                        <div><a href="#">Tech</a></div>
+                        <h1>Can dark matter be found on<br/> earth? New research says ‘maybe’</h1>
+                        <ul>
+                            <li>
+                                by
+                                <a href="#" class="link-white">The Cosmic Companion</a>
+                            </li>
+                            <li><a href="#" class="link-white"><i class="far fa-clock"></i> 10 hours ago</a></li>
+                            <li>
+                                <i class="fas fa-share-alt link-white"></i> 0
+                            </li>
+                        </ul>
```
We have to remove all the commented codes. So, remove all the comments from the HTML and CSS file.

sipe-daniel replied:
ok, I removed these comments

sumancrest0001 commented on index.html:
```diff
@@ -1,9 +1,543 @@
-<html>
+<!DOCTYPE html>
+<html lang="en">
     <head>
+        <title>TNW</title>
```
Let's remove the HTML validator error. I am using [this validator](https://validator.w3.org/#validate_by_upload)

sumancrest0001 commented on css/style.css:
```diff
@@ -0,0 +1,1027 @@
+  margin-left: 1.5em;
+}
+
+@media (max-width: 935px) {
+  nav.wrapperleft-link > ul {
+    display: none;
+  }
+
+  nav.wrapperright-link > ul {
+    display: none;
+  }
+
+  div.headersecond-row > div > p,
+  div.headersecond-row > nav > ul > li:nth-child(2),
+  div.headersecond-row > nav > ul > li:nth-child(3),
+  div.headersecond-row > nav > ul > li:nth-child(4),
+  div.headersecond-row > nav > ul > li:nth-child(5),
+  div.headersecond-row > nav > ul > li:nth-child(6),
+  div.headersecond-row > nav > ul > li:nth-child(7),
+  div.headersecond-row > nav > ul > li:nth-child(8) {
+    display: none;
+  }
```
We have to remove the repeated code and follow the DRY principle. Instead of using such long selector (  div.headersecond-row > nav > ul > li:nth-child(2),), we can define class with the required styling and add where ever we need. Another option is we can we `not` selector. checkout this [article](https://speckyboy.com/good-bad-css-practices)

sipe-daniel replied:
I modified this part.

sumancrest0001 commented on css/style.css:
```diff
@@ -0,0 +1,1027 @@
+  }
+}
+
+.wrappersubgrid > div > div:nth-child(1) {
+  background-image: url("https://cdn0.tnwcdn.com/wp-content/blogs.dir/1/files/2020/05/Sprint_image.jpg");
+  background-size: cover;
+  height: 11.5em;
+  width: 98%;
+  margin-left: 0.5em;
+  margin-bottom: 0.3em;
+  margin-top: 0.1em;
+}
+
+@media (max-width: 600px) {
+  .wrappersubgrid > div > div:nth-child(1) {
+    background-image: url("https://cdn0.tnwcdn.com/wp-content/blogs.dir/1/files/2020/05/Sprint_image.jpg");
+    background-size: cover;
+    height: 11.5em;
+    width: 96%;
+    margin-left: 0.5em;
+    margin-bottom: 0.3em;
+    margin-top: 0.1em;
+  }
+}
+
```
Let's remove the repeated code and make the CSS file clean.

sipe-daniel replied:
ok thanks, I resolved this

sumancrest0001 commented on css/style.css:
```diff
@@ -0,0 +1,1027 @@
+    font-size: 28px;
+  }
+}
+
+.second-content .wrappersubgrid-two > .first-case > img,
+.second-content .wrappersubgrid-two > .second-case > img,
+.second-content .wrappersubgrid-two > .third-case > img,
+.second-content .wrappersubgrid-two > .fourth-case > img,
+.second-content .wrappersubgrid-two > .fifth-case > img,
+.second-content .wrappersubgrid-two > .sixth-case > img,
+.second-content .wrappersubgrid-two > .seventh-case > img,
+.second-content .wrappersubgrid-two > .eighth-case > img {
+  height: 160px;
+  width: 96%;
+}
+
+.second-content .wrappersubgrid-two > .first-case > h3,
+.second-content .wrappersubgrid-two > .second-case > h3,
+.second-content .wrappersubgrid-two > .third-case > h3,
+.second-content .wrappersubgrid-two > .fouth-case > h3,
+.second-content .wrappersubgrid-two > .fifth-case > h3,
+.second-content .wrappersubgrid-two > .sixth-case > h3,
+.second-content .wrappersubgrid-two > .seventh-case > h3,
+.second-content .wrappersubgrid-two > .eighth-case > h3 {
+  text-align: initial;
```
Let's use class instead of long selectors.

sipe-daniel replied:
ok, thanks. I changed the selectores

sumancrest0001 commented:
Since it is a clone project. We have to create a similar layout as that of the original one. Let's add titles and other details to each article.
currently :point_down: 
![Screenshot from 2020-05-20 07-51-32](https://user-images.githubusercontent.com/37747619/82398535-cb8a2900-9a72-11ea-83da-0f656cb3aa59.png)

It should be like this :point_down: 
![Screenshot from 2020-05-20 07-51-50](https://user-images.githubusercontent.com/37747619/82398560-d644be00-9a72-11ea-997a-625dc2811a1f.png)

sumancrest0001 commented:
Let's give similar to the latest news section and change the styling of the **latest news** title.
In your project :point_down: 
![Screenshot from 2020-05-20 07-52-10](https://user-images.githubusercontent.com/37747619/82398729-3a678200-9a73-11ea-9ada-60d4d06ae7d5.png)


in the original project :point_down: 
![Screenshot from 2020-05-20 07-52-31](https://user-images.githubusercontent.com/37747619/82398751-481d0780-9a73-11ea-90a7-c3ca5520011b.png)

sumancrest0001 commented:
The category section should have 3 columns but currently, it is two columned. change the color of the latest sub-section and add bullet points as in the original page.
Have a look :point_down: 
![Screenshot from 2020-05-20 07-52-54](https://user-images.githubusercontent.com/37747619/82398897-9e8a4600-9a73-11ea-97bb-965ec014a052.png)

In the original site :point_down: 
![Screenshot from 2020-05-20 07-53-16](https://user-images.githubusercontent.com/37747619/82398916-aa760800-9a73-11ea-9ba3-d7bfb3a004d1.png)

sumancrest0001 commented:
It seems like you forgot to add a title to the **latest deal** section. Please add that.
![Screenshot from 2020-05-20 07-53-59](https://user-images.githubusercontent.com/37747619/82399046-fb85fc00-9a73-11ea-9baa-97fa706f9e36.png)

sumancrest0001 reviewed (changes requested):
Please make the requested changes and make the page responsive accordingly on the medium and small screens. Then request another review.
Need help or have question? DM me on slack **Suman Shrestha**

**Happy coding !!!**

sipe-daniel committed:
perf(principal-page): add codes and fix some errors

add sub-section, change the styling of the latest news title, add titles and other details to each article

trekab reviewed (changes requested):
# Changes Required!
Hi @sipe-daniel. Thank you for working on the feedback from the last review. However, there are several changes you need to make to this project to have it approved. These include; 

### Navigation:
- [x] TWN logo appears at the leftmost part of both the main and sub-navigation rows for medium screens. ([link to page](https://thenextweb.com/))
- [x] For small screens, your navigation is missing a search icon and the TNW logo has unnecessary white space on its left.
### Top 3 articles section:
- [x] The main article on the left should be wider (around two-thirds of the width) than the right two articles for large screens and medium screens.
- [x] It should have a TWN logo at the top left corner of the row for large screens. ([link to page](https://thenextweb.com/))
### Latest news section:
- [x] It aligns 3 articles per row for medium screens. ([link to page](https://thenextweb.com/))
- [x] It aligns 1 article per row for small screens and each article has a cover image on the left side and title on the right side of the row ([link to page](https://thenextweb.com/))
### Latest Funding section:
- [x] It aligns 3 articles per row for medium screens. ([link to page](https://thenextweb.com/))
### Latest deals section:
- [x] This section should behave like the `Latest news` section(above) for all breakpoints. ([link to page](https://thenextweb.com/))
### About media queries:
- [x] Your CSS styles should use [media queries](https://www.w3schools.com/css/css_rwd_mediaqueries.asp) for only 2 breakpoints ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/responsive_design/breakpoints_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59540217/)). 
- [x] You should have no small breakpoint as it is mobile-first and the base styles are for the mobile screen. ([Screenshot on mobile-first approach](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/responsive_design/mobile_first.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59540217/))
- [x] medium: min-width around 768px
- [x] large: min-width 1024px

__N:B:__ All these requirements are mandatory.
__Kindly make the necessary changes and submit another code review request__
_Feel free to reach me on slack for any inquiries about this code review._ :wink:
__Treasure Kabareebe__

sipe-daniel committed:
refactor(principal-page): change of alignment for some elements

TWN logo was appearing at the leftmost part of both the main and sub-navigation rows for medium screens, some problem with the alignment of some articles about sections Lastest news section, Latest Funding section, Latest deals section

sipe-daniel committed:
refactor(principal-page): improve the section before the Lastest deals section

improve @media for small screens

sipe-daniel committed:
refactor(principal-page): improve the section before the Lastest deals section

improve @media for small screens

rammazzoti2000 reviewed (approved):
### :heavy_check_mark: Project Approved :tada: :rocket: 
Hello @sipe-daniel you've come  a long way since the beginning of this pull request, great job on implementing the changes required by tse, this project requires quite lots of effort and keen attention on small details, I see that you have tried your best, congrats on that.
Consider this project approved, but remember as previous TSEs said this project should be mobile first yours is a mixture of `Desktop First` and `Mobile First` and the css styling relies entirely on media queries.
Also take a moment to reflect on how your project looks and how it should've looked based on the [original website](https://thenextweb.com/), do they look the same :question: 
Since the next project is going to be the Bootstrap project, my advice is to prepare yourself in advance and follow tutorials that would help you with positioning and displaying elements on the page, bootstrap is a big project and very long and requires lots of effort, good HTML structure etc.
Wish you the very best :smile: 
###
<em>Cheers and Happy coding! :clap: 
Code review by Alexandru Bangau

(Feel free to contact me on slack by the handle of alexandru if you have any additional question)</em>
### :heavy_check_mark: Status: Project Approved
